### DISCUSSION TITLE
Cluster integration

### DISCUSSION THREAD

**Discussion by int2001** [magicbug/Cloudlog#2299] [Ideas / Feature Requests]
Currently working on the following Idea:

If you're sweeping over the bands you sometimes hear stations calling qrz, or stations in the middle of a QSO.
Would be nice to know, if they were spotted. So i always watch the dxcluster, if QRG matches. After that i know who is qrv there.

Idea was to integrate that into cloudlog. A small button at the "add QSO"-Form, which looks up the Cluster who was last spotted on that QRG. If match, then prefill the Call. Of cause this will be configurable.

There are a few services available via API where one is able to get that information. I'm also working on a small in-memory-db which collects the last 1000 (or so) spots and make them searchable.

What do you think about that feature?

**Comment by dg9vh** [comment]
Hi, maybe it would be also a nice idea to have something like a bandmap, where the spots are listed by qrg.

**Comment by magicbug** [comment]
Bandmap (similar to those seen in contest apps) makes the most sense probably in a separate browser window, I did provide a dxcluster API for incoming spots but it largely got abused so shut it down, that said could revive for the interests of Cloudlog

**Comment by int2001** [comment]
i wrote a small dxcluster to REST Script --> https://github.com/int2001/DXClusterAPI
You can try that out here (Caches last 1000 spots): https://log.jo30.de/dxcache/spots

If your API is doing the same @magicbug, we also can use that, or make it configurable. Is there any documentation/playground of yours?

I think bandmap is also a good idea. But at first i'd like to start with that button from above.

**Reply by magicbug** [reply]
I did it a similar way using Perl script :-) so I'm happy to use anything new and host something permanent on Cloudlog/magicbug servers

**Reply by int2001** [reply]
try it. i'm also an "old-fashioned-Perl-Dev". But the last years i try node.js with pleasure.
Will create a PR for the integration, so you can test that. (@AndreasK79 already did ;)

If you need help with the node-thing, let me know.
i launched it on my server with pm2 and added routing via haproxy here.
You can also run it without haproxy on it's own port - as u like

**Reply by int2001** [reply]
forgot to publish a sample-config at the DXClusterAPI-Repo. Did that now.

**Reply by int2001** [reply]
@magicbug Some architectural considerations.
At present Cloudlog only holds the URL to the DXCluster-Cache to present the Endpoint to the User/JS when the User clicks on the loupe at "Add QSO". So there's no direct interaction between CL and the DXCluster-Cache.

Cloudlog has great functions to add more info to the Spot (already worked, Entity, Continent, etc.). Would be stupid to implement all those again (redundancy) to the DXCluster-Cache.

So following Idea: Let CL act as a proxy. Whenver User wants Cluster-Informations, Cloudlog is called in the Future. Cloudlog leeches the Info itself out of the DXCluster-Cache and adds all needed Information, mentioned above, and provide that back to the logged-in User.

What do you think?

**Reply by magicbug** [reply]
Yes seems to make sense, I was considering whether we should feed cluster spots to users via a web socket maybe something to look at further down the road. 

Be good if user could select what Cloudlog checks for but maybe that's going to far :)

**Comment by int2001** [comment]
Very early beta, just to show the direction.

In the Video you can see the following things:
- QSY at TRX to QRG
- see how CL takes QRG by CAT
- click on label (will change that to an icon later)
- clustercache is asked for latest spot at QRG. Spotted call will be filled to Callsign field, if there's a spot 

- clustercache-url configurable via config.php (leaving it blank disables the feature)

https://github.com/magicbug/Cloudlog/assets/1410708/508ee773-7a88-4681-b809-2aca14f93502

**Comment by int2001** [comment]
Early stage of a bandmap, based on the DXCache

https://github.com/magicbug/Cloudlog/assets/1410708/105d414e-25d8-47c0-bcde-57b3d3fe5eea

still needs some work.
Question to @magicbug : If finished (i will work with @AndreasK79 on that): where should we put it?

**Reply by magicbug** [reply]
Oh, I do like that, I think it needs to be a popout separate window personally

**Reply by magicbug** [reply]
Guessing its possible to show the rigs frequency on the bandmap too

**Reply by int2001** [reply]
sure. already had that on my todolist.
currently working on live-updates of the map.

**Reply by int2001** [reply]
Selfupdating: ✓
Age: ✓
Vertical: ✓
6 Columns wide instead of one or two: ✓

Still a lot to do. But i like

https://github.com/magicbug/Cloudlog/assets/1410708/fe5d9de9-e1a5-49db-a479-cb8fddf2da42

Working branch here: https://github.com/int2001/bandmap/tree/vertical
will get that a bit better, before i implement that into CL

**Comment by int2001** [comment]
I'm not really longer sure if those graphical approach is the right one.
With the libs we're using there are still some problems with overlapping, readability and so on.
As u can see the Alpha-Version is working, including selfupdating (00:12 in the Video) and CAT-support (not on screen). Some filters are also included. You can set up the max-age at the settings. This all has been done during the last few days by @AndreasK79 and me. Was really fun working together here.

But we're not really happy about the design. Yes it may look awesome, but is it usable?
Personally i'm using DXHeat a lot. And i like their simple UI with the list of spots as list.

But i'm also missing some features there. Like sorting by QRG, showing/highlighting at what QRG i am with my radio and so on. i think this gap can be filled by a cloudlog-integration.

Before we invest more time to the graphical approach. What do you think?

https://github.com/magicbug/Cloudlog/assets/1410708/98fc0bcd-ef6d-4516-89ea-80997af30e40

**Reply by magicbug** [reply]
For the interface i was thinking completely floating window so wouldn't need the header and could also allow it to keep it quite narrow

**Reply by int2001** [reply]
do you know a lib for that? Problem is overlapping. and i'm really not sure if this approach is more eyecandy than usable.

**Comment by dg9vh** [comment]
Looks Goodyear I think.

Joerg ***@***.***> schrieb am Fr., 21. Juli 2023, 17:17:
